slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -279,7 +279,80 @@ ::grpc::Status SLAMServiceImpl::GetCurrentPointCloudMap(
 ::grpc::Status SLAMServiceImpl::GetPointCloudMapStream(
     ServerContext *context, const GetPointCloudMapStreamRequest *request,
     ServerWriter<GetPointCloudMapStreamResponse> *writer) {
-    return ::grpc::Status(::grpc::StatusCode::UNIMPLEMENTED, "");
+    std::string pointcloud_map;
+    int num_points;
+    // Write or grab the latest pointcloud map in form of a string
+    try {
+        std::shared_lock optimization_lock{optimization_shared_mutex,
+                                           std::defer_lock};
+        if (optimization_lock.try_lock()) {
+            // We are able to lock the optimization_shared_mutex, which means
+            // that the optimization is not ongoing and we can grab the newest
+            // map
+            num_points = GetLatestSampledPointCloudMapString(pointcloud_map);
+            std::lock_guard<std::mutex> lk(viam_response_mutex);
+            latest_pointcloud_map = pointcloud_map;
```

Comment:
set `latest_num_points` here and in `BackupLatestMap()` and `CacheMapInLocalizationMode()`